core/src/test/java/org/elasticsearch/bootstrap/BootstrapCheckTests.java

```diff
@@ -197,4 +199,15 @@ public void testEnforceLimits() {
         assertTrue(BootstrapCheck.enforceLimits(settings));
     }
 
+    public void testMinMasterNodes() {
+        BootstrapCheck.Check check = new BootstrapCheck.MinMasterNodesCheck();
+        assertTrue(check.check(Settings.EMPTY));
+        assertFalse(check.check(
+            Settings.builder().put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 1).build()
+        ));
+        expectThrows(RuntimeException.class, () ->
+            BootstrapCheck.check(
```

Comment:
There's a `Bootstrap#check(boolean, Settings)` override that is visible for testing exactly so that the test can be written without having to rely on passing in a setting the triggers enforcement.